python/pyspark/sql/tests/test_python_streaming_datasource.py

```diff
@@ -295,6 +296,156 @@ def test_stream_writer(self):
             output_dir.cleanup()
             checkpoint_dir.cleanup()
 
+    def test_stream_arrow_writer(self):
+        """Test DataSourceStreamArrowWriter with Arrow RecordBatch format."""
+        import tempfile
+        import shutil
+        import json
+        import os
+        import pyarrow as pa
+        from dataclasses import dataclass
+
+        @dataclass
+        class ArrowCommitMessage(WriterCommitMessage):
+            partition_id: int
+            batch_count: int
+            total_rows: int
+
+        class TestStreamArrowWriter(DataSourceStreamArrowWriter):
+            def __init__(self, options):
+                self.options = options
+                self.path = self.options.get("path")
+                assert self.path is not None
+
+            def write(self, iterator):
+                from pyspark import TaskContext
+
+                context = TaskContext.get()
+                partition_id = context.partitionId()
+                batch_count = 0
+                total_rows = 0
+
+                for batch in iterator:
```

Comment:
This is actually odd - from write_into_data_source.py, I guess we pass `Iterator[Iterator[RecordBatch]]` and here we get `Iterator[RecordBatch]`. It sounds to me there is a magic happening somewhere.